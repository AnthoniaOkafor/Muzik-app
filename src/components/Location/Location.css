.locationName {
    margin: 15px 65px 40px 65px;
}

.nameClass {
    color: #1abc9c;
    font-size: 28px;
}

.locationClass {
    color: #7F8C8D;
    font-size: 18px;
    margin-top: 10px;
}

.fa-map-marker { 
    margin-right: 15px;
}