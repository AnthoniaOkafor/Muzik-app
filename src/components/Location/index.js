import React from "react";
import "./Location.css";
import PropTypes from "prop-types";


const Location = (props) => {
    const {name, location} = props;

    return (
        <div className = "locationName">
            <p className="nameClass">{name}</p>
            <p className="locationClass"><i className="fas fa-map-marker"></i>{location}</p>
        </div>
    )
};

Location.defaultProps = {
    name: "Name",
    location: "Location"
};

Location.PropTypes = {
    name: PropTypes.string,
    location: PropTypes.string
};


export default Location;