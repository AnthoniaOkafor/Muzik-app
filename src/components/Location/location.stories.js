import React from "react";
import {storiesOf} from "@storybook/react";
import {withInfo} from "@storybook/addon-info";
import {withKnobs, text} from "@storybook/addon-knobs";
import Location from "./index";

storiesOf("Location", module)
    .addDecorator(
        withInfo({
            text: "This is a Location component"
        })
    )

    .addDecorator(
        withKnobs()
    )

    .add("Location", () => (
        <Location
            name={text("name", "Event venue")}
            location={text("location","event location")} 
        />
    ));