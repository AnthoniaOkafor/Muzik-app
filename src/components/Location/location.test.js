import React from "react";
import {shallow} from "enzyme";
import Location from "./index";



describe ("Location Component render", () => {
   it("renders correctly", () => {
       const wrapper = shallow(<Location name="The name of the event" />);
       expect(wrapper).toMatchSnapshot();
    })

    it("renders correctly", () => {
        const wrapper = shallow(<Location location="The location of the event" />);
        expect(wrapper).toMatchSnapshot();
     })

     it ("renders font awesome icon", () => {
         const wrapper = shallow(<Location />);
            const faIcon = wrapper.find("i");
            expect(faIcon.length).toBe(1);
     })

     it("renders text on the page", () => {
        const wrapper = shallow(<Location />);
            const findText = wrapper.find("p");
            expect(findText.length).toBe(2)
     })

     it("renders correct text", () => {
      const wrapper = shallow(<Location name="The name of the event" location="The location of the event" />);
      expect(wrapper.text()).toBe("The name of the eventThe location of the event");
      console.log(wrapper.debug());
     })
    
      
});