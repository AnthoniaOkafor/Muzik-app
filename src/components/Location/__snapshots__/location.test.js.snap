// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Component render renders correctly 1`] = `ShallowWrapper {}`;

exports[`Location Component render renders correctly 2`] = `ShallowWrapper {}`;
